src/sentry/rules/processing/delayed_processing.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from collections.abc import Mapping
+
+from sentry.buffer.redis import BufferHookEvent, RedisBuffer, redis_buffer_registry
+from sentry.models.project import Project
+from sentry.utils import metrics
+from sentry.utils.query import RangeQuerySetWrapper
+from sentry.utils.safe import safe_execute
+
+logger = logging.getLogger("sentry.rules.delayed_processing")
+
+
+PROJECT_ID_BUFFER_LIST_KEY = "project_id_buffer_list"
+
+
+@redis_buffer_registry.add_handler(BufferHookEvent.FLUSH)
+def process_delayed_alert_conditions(buffer: RedisBuffer) -> None:
+    with metrics.timer("delayed_processing.process_all_conditions.duration"):
+        project_ids = buffer.get_set(PROJECT_ID_BUFFER_LIST_KEY)
```

Comment:
When do we delete all these keys from redis?

Reply:
Good point - I should make a method to do this. I think I handled cleaning out the hash during processing but missed the project ids.